components/candle/challengeService.ts

```diff
@@ -1274,6 +1274,26 @@ export class ChallengeService extends ChallengeRegistry {
                     gameVersion,
                     unlockableIds,
                 )
+
+                // If missions type is evergreen, checks if any of the unlockables has unlockable gear, and award those too
+                if (contract.Metadata.Type === "evergreen") {
```

Comment:
I'm pretty sure this is not how it works. Is this how it works. If you get to mastery level 5 of freelancer, the "EG_MASTERY_SKIN_LUXURIOUS_PISTOL" will be unlocked. This is not tied to any challenge in `_SNUG_CHALLENGES.json` - it just happens. So adding logic to handle this in challenges doesn't actually do anything. @moonysolari - thoughts? I feel we should add an optional "Drops" field to the Mastery Json type and check on that.

So:
```
{
  "Id": "EG_MASTERY_SKIN_LUXURIOUS_RIFLE",
  "Level": 10
}
```

Would become:
```
{
  "Id": "EG_MASTERY_SKIN_LUXURIOUS_RIFLE",
  "Level": 10,
  "Drops": [dropID]
}
```

Reply:
Everything in the mastery of Freelancer has an unlockable attached to it. All of them can be added to the inventory.

Reply:
There is also the "Gauze Suit" that I feel drops into your inventory after you die in Freelancer for the first time. So that's a whole other thing, again.

Reply:
I think its just a consequence of this being on the challenges as we spoke before. As @AnthonyFuller mentioned, the gear unlocks from freelancer have an unlock id for the normal inventory, which is not really tied to the challenges, it just feels weird because where this is being done. We can move this to the progression service once that's done, suppose it will make more sense there.

Not sure about the suit tho, can investigate, but supposed its tied to a challenge that its not tracked yet, we would have to look into that first

Reply:
The Gauze Suit is a drop from a challenge.

Reply:
> I feel we should add an optional "Drops" field to the Mastery Json type and check on that.

Sorry I'm a bit confused about why this needs to be added. I just tested and I can unlock The Ornamental Pistol on freelancer level 5. It appears in both the safehouse and in my inventory outside of freelancer, and I can take it to normal missions. Failing a mission also grants The Gauze Suit in my inventory. The items were also correctly locked before they were unlocked.

Reply:
I think it may just be some confusion around the need for this. If it helps clarify the logic, effectively the ID for the freelancer version of the weapon is not the same as the one for the normal inventory, so unlocking only EG_MASTERY_SKIN_LUXURIOUS_RIFLE as per snug mastery list , will not unlock it on the inventory (also the reason why it was unlocked to begin with, because the inventory version was not behind mastery progression and so awarded to the user by default). 

Each evergreen gear type unlockable has an `Unlocks` field that references the ID for the inventory version, hence why the need to drill into the unlockables to unlock both version.

Reply:
> Each evergreen gear type unlockable has an Unlocks field that references the ID for the inventory version, hence why the need to drill into the unlockables to unlock both version.

But I just tested and it seems both versions are already unlocked... Can you confirm?